### PR TITLE
Spring Boot practice

### PR DESCRIPTION
**Pull request by VasylAzarov**
add category entity | now books has category set | add category endpoints and documentation | add category mapper | update book entity, service, repo, mapper, controller | add new db scripts

### CODE REVIEW

File: src/main/java/dev/vasyl/proj/dto/book/CreateBookRequestDto.java [hunk 1]
```diff
@@ -32,4 +33,7 @@ public class CreateBookRequestDto {
     @Schema(description = "Cover image of the book",
             example = "https://cover-book-image.com/12")
     private String coverImage;
+    @Schema(description = "List of Category ids",
+            example = "[1, 2]")
+    private List<Long> categoryIds;
```
Comment (Elena-Bruyako) [inline src/main/java/dev/vasyl/proj/dto/book/CreateBookRequestDto.java:40]:
```suggestion
    @NotEmpty
    private List<Long> categoryIds;
```

File: src/main/java/dev/vasyl/proj/service/impl/CategoryServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto update(Long id, CreateCategoryRequestDto requestDto) {
+        Category existingCategory = categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find Category entity by id: " + id));
+        categoryMapper.updateCategoryFromDto(requestDto, existingCategory);
+        Category updatedCategory = categoryRepository.save(existingCategory);
+        return categoryMapper.toDto(updatedCategory);
```
Comment (Elena-Bruyako) [inline src/main/java/dev/vasyl/proj/service/impl/CategoryServiceImpl.java]:
```suggestion
        categoryRepository.save(existingCategory);
        return categoryMapper.toDto(existingCategory);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/dev/vasyl/proj/controller/CategoryController.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            description = "Get category by id. Available for User and Admin roles")
+    public CategoryDto getCategoryById(@PathVariable Long id) {
+        return categoryService.findById(id);
+    }
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @PostMapping
+    @Operation(summary = "Create a new category",
+            description = "Create a new category. Available for Admin role")
+    public CategoryDto createCategory(@RequestBody @Valid CreateCategoryRequestDto requestDto) {
```
Comment (JJJazl) [inline src/main/java/dev/vasyl/proj/controller/CategoryController.java:60]: Let's add a corresponding status code for POST method

File: src/main/java/dev/vasyl/proj/repository/BookRepository.java [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
 
+@Repository
 public interface BookRepository extends JpaRepository<Book, Long> {
 
     boolean existsByIsbn(String isbn);
 
+    Page<Book> findByCategoriesId(Long categoryId, Pageable pageable);
```
Comment (JJJazl) [inline src/main/java/dev/vasyl/proj/repository/BookRepository.java:14]:
```suggestion
    Page<Book> findByCategoryId(Long categoryId, Pageable pageable);
```

**Review by JJJazl — APPROVED** [review]

**Pull request merged**